aws/resource_aws_ssm_parameter.go

```diff
@@ -201,76 +261,70 @@ func resourceAwsSsmParameterRead(d *schema.ResourceData, meta interface{}) error
 	return nil
 }
 
-func resourceAwsSsmParameterDelete(d *schema.ResourceData, meta interface{}) error {
-	ssmconn := meta.(*AWSClient).ssmconn
+func resourceAwsSsmParameterUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ssmconn
+
+	if d.HasChangesExcept("tags") {
+		paramInput := &ssm.PutParameterInput{
+			Name:           aws.String(d.Get("name").(string)),
+			Type:           aws.String(d.Get("type").(string)),
+			Tier:           aws.String(d.Get("tier").(string)),
+			Value:          aws.String(d.Get("value").(string)),
+			Overwrite:      aws.Bool(shouldUpdateSsmParameter(d)),
+			AllowedPattern: aws.String(d.Get("allowed_pattern").(string)),
+		}
 
-	log.Printf("[INFO] Deleting SSM Parameter: %s", d.Id())
+		if d.HasChange("data_type") {
```

Comment:
..turns out you can't actually change a "data_type" tho (i tried going from image -> text, but not vice versa yet)

```
Error: error updating SSM parameter (tf-acc-test-1338304779148665738): ValidationException: You can’t change a parameter’s data type.
        	status code: 400, request id: d166482e-29ef-4f9c-822d-6c8a0f0f99ca
```